engine/src/main/java/org/hibernate/validator/messageinterpolation/ResourceBundleMessageInterpolator.java

```diff
@@ -21,32 +23,54 @@
  * @author Adam Stawicki
  */
 public class ResourceBundleMessageInterpolator extends AbstractMessageInterpolator {
+	private final ExpressionFactory expressionFactory;
+
+	// Many constructors here are only called for tests; these just call ExpressionFactory.newInstance().
+	// Some sort of DI would be neater :-)
+
 	public ResourceBundleMessageInterpolator() {
 		super();
+		this.expressionFactory = ExpressionFactory.newInstance();
 	}
 
-	public ResourceBundleMessageInterpolator(ResourceBundleLocator userResourceBundleLocator) {
+	public ResourceBundleMessageInterpolator(ResourceBundleLocator userResourceBundleLocator,
```

Comment:
We must not alter these constructors (it's public facing API). Could you instead add a new one, please? Admittedly, the number of constructors gets unwieldy, probably we should add some kind of builder eventually.